app/gui2/src/util/ast/match.ts

```diff
@@ -44,7 +45,9 @@ export class Pattern {
     for (const matched of placeholders(ast, this.placeholder)) {
       const replacement = subtrees.shift()
       if (replacement === undefined) break
-      edit.get(replacement)!.parent = matched.parent
+      const replacementAst = edit.get(replacement)
```

Comment:
Should we refactor the function to not take a `MutableModule`, too? Since we want to ensure it is not modified.

Reply:
We are making a modified copy of `replacement` in the `edit` module. Before this change, we would sometimes mutate it in its own module instead. We don't have a `MutableModule` reference to `replacement`'s module, but TS doesn't give us any way to enforce immutability.